server/src/main/java/org/apache/iotdb/db/metadata/schemaregion/SchemaEngine.java

```diff
@@ -143,10 +160,37 @@ public synchronized void createSchemaRegion(
     if (schemaRegion != null) {
       return;
     }
-    localStorageGroupSchemaManager.ensureStorageGroup(storageGroup);
+    createSchemaRegionWithoutExistenceCheck(storageGroup, schemaRegionId);
+  }
+
+  private Runnable createSchemaRegionTask(PartialPath storageGroup, SchemaRegionId schemaRegionId) {
+    // this method is called for concurrent recovery of schema regions
+    return () -> {
+      long timeRecord = System.currentTimeMillis();
+      ISchemaRegion schemaRegion = this.schemaRegionMap.get(schemaRegionId);
+      if (schemaRegion != null) {
+        return;
+      }
+      try {
+        createSchemaRegionWithoutExistenceCheck(storageGroup, schemaRegionId);
+        timeRecord = System.currentTimeMillis() - timeRecord;
+        logger.info(
+            String.format(
+                "Recover [%s] spend: %s ms",
+                storageGroup.concatNode(schemaRegionId.toString()), timeRecord));
+      } catch (MetadataException e) {
+        throw new RuntimeException(e);
+      }
+    };
+  }
+
+  public void createSchemaRegionWithoutExistenceCheck(
+      PartialPath storageGroup, SchemaRegionId schemaRegionId) throws MetadataException {
+    ISchemaRegion schemaRegion = null;
+    this.localStorageGroupSchemaManager.ensureStorageGroup(storageGroup);
```

Comment:
Why we provide this as a public interface? It seems a creation with existence check is better to be provided.

Reply:
fixed now